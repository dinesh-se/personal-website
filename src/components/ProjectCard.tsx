import Link from "next/link";

import { RepoUI } from "../types/github";

const ProjectCard = ({
  name,
  description,
  href,
  language
}: RepoUI) => {
  return (
    <Link className="bg-neutral-300 dark:bg-slate-900 rounded-md overflow-hidden p-6" href={href}>
      <h2 className="font-semibold mb-2 text-ellipsis text-nowrap overflow-hidden">{name}</h2>
      <p className="text-gray-600 text-sm">{description}</p>
      <p className="mt-4 text-sm">
        <span className="font-semibold">Tech:&nbsp;</span>
        {language}
      </p>
    </Link>
  )
}

export default ProjectCard;
