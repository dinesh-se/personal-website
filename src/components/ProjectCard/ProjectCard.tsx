import Link from 'next/link';

import { RepoUI } from '@types';

const ProjectCard = ({ name, description, href, primaryLanguage }: RepoUI) => (
	<Link
		className="overflow-hidden rounded-md bg-neutral-300 p-6 dark:bg-slate-900"
		href={href}
	>
		<h2 className="mb-2 overflow-hidden text-ellipsis text-nowrap font-semibold">
			{name}
		</h2>
		<p className="text-sm text-gray-600 dark:text-gray-300">{description}</p>
		{primaryLanguage && (
			<p className="mt-4 text-sm flex items-center">
				<span>{primaryLanguage.name}</span>
				<span
					className="mr-2 w-4 h-4 rounded-full order-first"
					style={{ backgroundColor: primaryLanguage.color }}
				/>
			</p>
		)}
	</Link>
);

export default ProjectCard;
