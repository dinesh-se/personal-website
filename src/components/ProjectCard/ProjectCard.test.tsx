import { render } from '@testing-library/react';

import { RepoUI } from '@types';

import ProjectCard from './ProjectCard';

describe('ProjectCard', () => {
	it('renders correctly', () => {
		const props: RepoUI = {
			name: 'test-repo-name',
			description: 'test-description',
			href: 'https://test.repo.url',
			primaryLanguage: {
				name: 'TypeScript',
				color: '#ff00ff',
			},
		};
		const { container } = render(<ProjectCard {...props} />);
		expect(container).toMatchSnapshot();
	});
});
