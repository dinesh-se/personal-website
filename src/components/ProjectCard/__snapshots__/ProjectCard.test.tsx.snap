// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectCard renders correctly 1`] = `
<div>
  <a
    class="overflow-hidden rounded-md bg-neutral-300 p-6 dark:bg-slate-900"
    href="https://test.repo.url"
  >
    <h2
      class="mb-2 overflow-hidden text-ellipsis text-nowrap font-semibold"
    >
      test-repo-name
    </h2>
    <p
      class="text-sm text-gray-600"
    >
      test-description
    </p>
    <p
      class="mt-4 text-sm"
    >
      <span
        class="font-semibold"
      >
        Tech:Â 
      </span>
      TypeScript
    </p>
  </a>
</div>
`;
